doc/classes/EditorUndoRedoManager.xml

```diff
@@ -0,0 +1,118 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<class name="EditorUndoRedoManager" inherits="RefCounted" version="4.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="../class.xsd">
+	<brief_description>
+		Manages undo history of scenes opened in the editor.
+	</brief_description>
+	<description>
+		[EditorUndoRedoManager] is a manager for [UndoRedo] objects associated with edited scenes. Each scene has its own undo history and EditorUndoRedoManager that each action performed in the editor gets associated with a proper scene. For actions not related to scenes ([ProjectSettings] edits, external resources etc.), a separate global history is used.
+		The usage is mostly the same as [UndoRedo]. You create and commit actions and the manager automatically decides under-the-hood what scenes it belongs to. The scene is deduced based on the first operation in an action, using the object from operation. The rules are as follows:
+		- if the object is a [Node], use the currently edited scene;
+		- if the object is a built-in resource, use the scene from its path;
+		- if the object is external resource or anything else, use global history.
+		This guessing can sometimes yield false results, so you can provide a custom context object when creating an action.
+	</description>
+	<tutorials>
+	</tutorials>
+	<methods>
+		<method name="add_do_method" qualifiers="vararg">
+			<return type="void" />
+			<param index="0" name="object" type="Object" />
+			<param index="1" name="method" type="StringName" />
+			<description>
+				Register a method that will be called when the action is committed.
+				If this is the first operation, the [param object] will be used to deduce target undo history.
+			</description>
+		</method>
+		<method name="add_do_property">
+			<return type="void" />
+			<param index="0" name="object" type="Object" />
+			<param index="1" name="property" type="StringName" />
+			<param index="2" name="value" type="Variant" />
+			<description>
+				Register a property value change for "do".
+				If this is the first operation, the [param object] will be used to deduce target undo history.
+			</description>
+		</method>
+		<method name="add_do_reference">
+			<return type="void" />
+			<param index="0" name="object" type="Object" />
+			<description>
+				Register a reference for "do" that will be erased if the "do" history is lost. This is useful mostly for new nodes created for the "do" call. Do not use for resources.
+			</description>
+		</method>
+		<method name="add_undo_method" qualifiers="vararg">
+			<return type="void" />
+			<param index="0" name="object" type="Object" />
+			<param index="1" name="method" type="StringName" />
+			<description>
+				Register a method that will be called when the action is undone.
```

Comment:
```suggestion
				Register a method that will be called when the action is undone (i.e. the "undo" action).
```
Not sure if `"do"/"undo" action` is the best term for this but I felt this needs to be defined once since other methods just refer to `"do"` and `"undo"` without defining them.

Reply:
Why this one says "action", but the other one "path".

Reply:
I edited it after sending the first, as I figured "path" would be tricky to translate for some languages.